src/server.c

```diff
@@ -1597,7 +1595,7 @@ void beforeSleep(struct aeEventLoop *eventLoop) {
     /* Write the AOF buffer on disk,
      * must be done before handleClientsWithPendingWritesUsingThreads,
      * in case of appendfsync=always. */
-    if (server.aof_state == AOF_ON || server.aof_state == AOF_WAIT_REWRITE)
+    if (server.aof_state == AOF_ON)
```

Comment:
Question: Does this change mean that we'll be buffering data in `aof_buf` until the rewrite is done? If that's the case the we loose one major advantage of mp-aof which is to avoid using up memory for buffering commands during rewrites. Shouldn't we write to the incr file during this time, but just make sure the manifest isn't updated until we're done with the rewrite?

Reply:
Yes, you are right (although this only happens when AOF is turned on, AOFRW executed dynamically does not have this problem), but your suggestion is good, I will work on it tomorrow, thanks.

Reply:
@yoav-steinberg Implemented, please review again, thanks.